llama_index/vector_stores/docarray/base.py

```diff
@@ -0,0 +1,181 @@
+
+        # check to see if empty document list was passed
+        if len(embedding_results) == 0:
+            return []
+
+        docs = DocList[self._schema](
+            self._schema(
+                id=result.id,
+                metadata=node_to_metadata_dict(result.node),
+                text=result.node.get_text(),
+                embedding=result.embedding,
+            )
+            for result in embedding_results
+        )
+        self._index.index(docs)
+        logger.info(f"Successfully added {len(docs)} documents to the index")
+        if self._ref_docs is not None:
+            self._update_ref_docs(docs)
+        return [doc.id for doc in docs]
+
+    def delete(self, doc_id: str, **delete_kwargs: Any) -> None:
+        """Deletes a document from the vector store.
+
+        Args:
+            doc_id (str): Document ID to be deleted.
+            **delete_kwargs (Any): Additional arguments to pass to the delete method.
+        """
+        docs_to_be_removed = self._find_docs_to_be_removed(doc_id)
+        if not docs_to_be_removed:
+            logger.warning(f"Document with doc_id {doc_id} not found")
+            return
+
+        del self._index[docs_to_be_removed]
+        logger.info(f"Deleted {len(docs_to_be_removed)} documents from the index")
+
+    def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
+        """Queries the vector store and retrieves the results.
+
+        Args:
+            query (VectorStoreQuery): Query for the vector store.
+
+        Returns:
+            VectorStoreQueryResult: Result of the query from vector store.
+        """
+        if query.filters:
+            # only for ExactMatchFilters
+            filter_query = {
+                "metadata__" + filter.key: {"$eq": filter.value}
+                for filter in query.filters.filters
+            }
+            query = (
+                self._index.build_query()  # get empty query object
+                .find(
+                    query=self._schema(embedding=np.array(query.query_embedding)),
+                    search_field="embedding",
+                    limit=query.similarity_top_k,
+                )  # add vector similarity search
+                .filter(filter_query=filter_query)  # add filter search
+                .build()  # build the query
+            )
+
+            # execute the combined query and return the results
+            docs, scores = self._index.execute_query(query)
+        else:
+            docs, scores = self._index.find(
+                query=self._schema(embedding=np.array(query.query_embedding)),
+                search_field="embedding",
+                limit=query.similarity_top_k,
+            )
+        nodes = [
+            Node(
+                doc_id=doc.id,
+                text=doc.text,
+                embedding=None,
+                relationships={
+                    DocumentRelationship.SOURCE: doc.metadata["doc_id"],
+                },
```

Comment:
could you revise to use `metadata_dict_to_node` to get back the various metadata for the node from the doc.metadata field?